janusgraph-test/src/test/java/org/janusgraph/core/ConfiguredGraphFactoryTest.java

```diff
@@ -0,0 +1,280 @@
+        try {
+            Map<String, Object> map = new HashMap<String, Object>();
+            map.put(STORAGE_BACKEND.toStringWithoutRoot(), "inmemory");
+            map.put(GRAPH_NAME.toStringWithoutRoot(), "graph1");
+            configGraphManagement.createConfiguration(new MapConfiguration(map));
+            StandardJanusGraph graph = (StandardJanusGraph) ConfiguredGraphFactory.open("graph1");
+            assertNotNull(graph);
+            assertEquals(graph.getConfiguration().getConfiguration().get(GRAPH_NAME), "graph1");
+            assertEquals(graph.getConfiguration().getConfiguration().get(STORAGE_BACKEND), "inmemory");
+        } finally {
+            configGraphManagement.removeConfiguration("graph1");
+            ConfiguredGraphFactory.close("graph1");
+        }
+    }
+
+    @Test
+    public void shouldCreateAndGetGraphUsingTemplateConfiguration() throws InterruptedException, BackendException {
+        try {
+            Map<String, Object> map = new HashMap<String, Object>();
+            map.put(STORAGE_BACKEND.toStringWithoutRoot(), "inmemory");
+            configGraphManagement.createTemplateConfiguration(new MapConfiguration(map));
+            StandardJanusGraph graph = (StandardJanusGraph) ConfiguredGraphFactory.create("graph1");
+            StandardJanusGraph graph1 = (StandardJanusGraph) ConfiguredGraphFactory.open("graph1");
+
+            assertNotNull(graph);
+            assertEquals(graph, graph1);
+        } finally {
+            configGraphManagement.removeConfiguration("graph1");
+            ConfiguredGraphFactory.close("graph1");
+        }
+    }
+
+    @Test
+    public void graphConfigurtionShouldBeWhatWeExpectWhenUsingTemplateConfiguration()
+        throws InterruptedException, BackendException {
+
+        try {
+            Map<String, Object> map = new HashMap<String, Object>();
+            map.put(STORAGE_BACKEND.toStringWithoutRoot(), "inmemory");
+            configGraphManagement.createTemplateConfiguration(new MapConfiguration(map));
+            StandardJanusGraph graph = (StandardJanusGraph) ConfiguredGraphFactory.create("graph1");
+            StandardJanusGraph graph1 = (StandardJanusGraph) ConfiguredGraphFactory.open("graph1");
+
+            assertNotNull(graph);
+            assertEquals(graph, graph1);
+            assertEquals(graph.getConfiguration().getConfiguration().get(GRAPH_NAME), "graph1");
+            assertEquals(graph.getConfiguration().getConfiguration().get(STORAGE_BACKEND), "inmemory");
+
+        } finally {
+            configGraphManagement.removeConfiguration("graph1");
+            ConfiguredGraphFactory.close("graph1");
+        }
+    }
+
+    @Test
+    public void shouldThrowConfigurationDoesNotExistError() {
+        boolean errorThrown = false;
+        try {
+            ConfiguredGraphFactory.open("graph1");
+        } catch (RuntimeException e) {
+            errorThrown = true;
+        }
+        assertTrue(errorThrown);
```

Comment:
There are much cleaner ways of testing for expected exceptions that don't require `try/catch` with a boolean variable and `assertTrue()`; see http://blog.codeleak.pl/2013/07/3-ways-of-handling-exceptions-in-junit.html for some examples; the code will become much shorter and easier to read with these changes.